apps/judicial-system/backend/src/app/modules/subpoena/subpoena.service.ts

```diff
@@ -320,4 +334,39 @@ export class SubpoenaService {
       return { delivered: false }
     }
   }
+
+  async getSubpoena(subpoena: Subpoena, user?: TUser): Promise<Subpoena> {
+    if (!subpoena.subpoenaId) {
+      // The subpoena has not been delivered to the police
+      return subpoena
+    }
+
+    if (subpoena.serviceStatus) {
+      // The subpoena has already been served to the defendant
+      return subpoena
+    }
+
+    // We don't know if the subpoena has been served to the defendant
+    // so we need to check the police service
+    const subpoenaInfo = await this.policeService.getSubpoenaStatus(
+      subpoena.subpoenaId,
+      user,
+    )
+
+    if (
```

Comment:
This if statement is a bit hard to read in my opinion. I believe it could be a bit cleaner with something like 

```
func isEqualOrUnset(fieldValue: keyof subpoenaInfo, checkValue: any) => {
    return !fieldValue || fieldValue === checkValue
}

const fieldsToCheck: keyof subpoeanInfo[] = [
  'serviceStatus',
  'comment',
  'servedBy',
  'defenderNationalId',
  'serviceDate',
]

if (fieldsToCheck.every((field) =>  isEqualOrUnset(subpoenaInfo[field], subpoena[field])))
```

By doing this you could also combine all three if statements into one.

Reply:
I want the conditions to be separate becuase they handle different scenarios and I think it is important to know why we are returning without updateing.

Reply:
I moved the diff calculation to the `SubpoenaInfo` class to make this code clearere.